packages/store-engine/src/main/engine/FileSystemEngine.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import CRUDEngine from './CRUDEngine';
+
+export default class FileSystemEngine implements CRUDEngine {
+  public storeName: string = '';
+
+  private filesystem: FileSystem | undefined;
+
+  constructor() {}
+
+  init(storeName: string = '', options: {type: number; size: number}): Promise<string> {
+    const config = Object.assign(
```

Comment:
Also, I would suggest adding a `FileSystemEngineOption` so that you can type the option in the parameters and the default options too.

This would give something like 

```javascript
interface FileSystemEngineOptions {
  type: number;
  size: number;
}
```

and then 

```init(/* ... */, options: FileSystemEngineOptions): Promise<string> {
  const defaultOptions: FileSystemEngineOptions = { ... };
  const config = { ...defaultOptions, ...options };
}

Reply:
agree